src/Mayaqua/Encrypt.c

```diff
@@ -4587,135 +4587,174 @@ void AesFreeKey(AES_KEY_VALUE *k)
 // AES encryption
 void AesEncrypt(void *dest, void *src, UINT size, AES_KEY_VALUE *k, void *ivec)
 {
-	UCHAR ivec_copy[AES_IV_SIZE];
 	// Validate arguments
 	if (dest == NULL || src == NULL || size == 0 || k == NULL || ivec == NULL)
 	{
 		return;
 	}
 
-#ifdef	USE_INTEL_AESNI_LIBRARY
-	if (is_intel_aes_supported)
+	// Create and initialize the context
+	EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+
+	if (!ctx)
 	{
-		AesEncryptWithIntel(dest, src, size, k, ivec);
+		ERR_print_errors_fp(stderr);
+		EVP_CIPHER_CTX_free(ctx);

```

Comment:
Since `ctx == NULL` here, there is no need to call EVP_CIPHER_CTX_free in this specific place. Same on the line 4679.

Reply:
Right.